plugins/org.eclipse.fordiac.ide.debug/src/org/eclipse/fordiac/ide/debug/fb/FBLaunchEventQueue.java

```diff
@@ -14,33 +14,52 @@
  *******************************************************************************/
 package org.eclipse.fordiac.ide.debug.fb;
 
+import java.time.Clock;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.ZoneId;
 import java.util.concurrent.BlockingQueue;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ConcurrentMap;
 import java.util.concurrent.LinkedBlockingQueue;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.concurrent.atomic.AtomicInteger;
 
+import org.eclipse.fordiac.ide.debug.EvaluatorProcess;
+import org.eclipse.fordiac.ide.model.eval.AbstractEvaluator;
 import org.eclipse.fordiac.ide.model.eval.fb.FBEvaluatorCountingEventQueue;
 import org.eclipse.fordiac.ide.model.eval.fb.FBEvaluatorExternalEventQueue;
 import org.eclipse.fordiac.ide.model.libraryElement.Event;
 
 public class FBLaunchEventQueue implements FBEvaluatorCountingEventQueue, FBEvaluatorExternalEventQueue {
 	private final AtomicBoolean repeat;
 	private final AtomicBoolean blocking;
+	private final AtomicBoolean system;
+	private final AtomicBoolean incremental;
+	private final AtomicBoolean manual;
```

Comment:
I would suggest defining an enum (and using an AtomicReference here), since these modes are mutually-exclusive. Otherwise, switching between modes may cause race conditions where more than one mode is `true` or all of them are `false`.